dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -753,38 +767,89 @@ private boolean validateServiceDescriptor(final AppSchema appDescriptor)
+            if (UtilMethods.isSet(descriptor.getHidden())
+                    && descriptor.isHidden()) {
+                errors.add(String.format(
+                        "Param `%s`: Bool params can not be marked hidden. The combination (Bool + Hidden) isn't allowed.",
+                        name));
+            }
+
+            if (UtilMethods.isSet(descriptor.getValue())
+                    && !isBoolString(descriptor.getValue().toString())) {
+                errors.add(String.format(
+                        "Boolean Param `%s` has a default value `%s` that can not be parsed to bool (true|false).",
+                        name, descriptor.getValue()));
+            }
         }
 
-        if (StringPool.NULL.equalsIgnoreCase(descriptor.getValue()) && descriptor.isRequired()) {
-            errors.add(String.format(
-                    "Null isn't allowed as the default value on required params see `%s`. ",
-                    name)
-            );
+        if(Type.STRING.equals(descriptor.getType())){
+            if(!(descriptor.getValue() instanceof String)){
+                errors.add(String.format(
+                        "Value Param `%s` has a default value `%s` that isn't a string .",
+                        name, descriptor.getValue()));
+            }
+        }
+
+        if (Type.SELECT.equals(descriptor.getType())) {
+
+            if (UtilMethods.isSet(descriptor.getHidden()) && descriptor.isHidden()) {
+                errors.add(String.format(
+                        "Param `%s`: List params can not be marked hidden. The combination (List + Hidden) isn't allowed.",
+                        name));
+            }
+
+            if (!(descriptor.getValue() instanceof List)) {
+                errors.add(String.format(
+                        " As param `%s`:  is marked as `List` the field value is expected to hold a list of objects. ",
+                        name));
+            } else {
+                int selectedCount = 0;
+                final List list = (List) descriptor.getValue();
+                for (final Object object : list) {
+                    if (!(object instanceof Map)) {
+                        errors.add(String.format(
+                                "Malformed list. Param: `%s` is marked as `List` therefore field `value` is expected to have a list of objects. ",
+                                name));
+                    } else {
+                        final Map map = (Map) object;
+                        if (!map.containsKey("label") || !map.containsKey("value") ) {
+                            errors.add(String.format("Malformed list. Param: `%s`. Every entry of the `List` has to have the following fields (`label`,`value`). ", name));
+                        }
+                         if(map.containsKey("selected")){
+                             selectedCount++;
+                         }
+                    }
+                }
+                if(selectedCount > 1 ){
```

Comment:
Codacy found an issue: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5895122)